lib/chef/knife/cookbook_upload.rb

```diff
@@ -34,10 +34,12 @@ class CookbookUpload < Knife
       banner "knife cookbook upload [COOKBOOKS...] (options)"
 
       option :cookbook_path,
-        short: "-o PATH:PATH",
-        long: "--cookbook-path PATH:PATH",
-        description: "A colon-separated path to look for cookbooks in.",
-        proc: lambda { |o| o.split(":") }
+        short: "-o 'PATH:PATH'",
+        long: "--cookbook-path 'PATH:PATH'",
+        description: "A colon-separated path to look for cookbooks in.
+                      Colon seprated path has been deprecated for Windows machine.
+                      In windows please use semicolon-separated path(For ex: 'PATH;PATH').",
```

Comment:
"A delimited path to search for cookbooks.  On Unix the delimiter is ':', on Windows it is ';'."

Reply:
@lamont-granquist Agreed, I updated the changes.